ourPLCC/plcc
# refactor!: convert plcc.py to python package plcc

```
refactor!: convert plcc.py to python package plcc

BREAKING CHANGE:

plcc is now a Python package, and plcc.py has been removed.
So plcc.py can no longer be called directly like this

    python3 path/to/plcc.py grammar

Instead, use plcc like this

    plcc grammar

---

This commit restructures the PLCC codebase into that of a
conventional Python project. This should make it easier to
redesign the internal structure of PLCC, making it easier to
maintain and add new functionality. It should also help us
prepare to distribute PLCC through PyPI, making it easier
to install. Here is a summary of changes.

- Convert plcc.py into python package plcc
- Remove environment variable LIBPLCC
- Move src/Std/ under src/plcc (the PLCC Python package)
- Move lib/jackson under src/plcc (the PLCC Python package)
- Pull common functionality of scripts into common.bash

---

Related Issues

- Related to #119
- Closes #121 
- Closes #120 
```

- Related to #119
- Closes #121 
- Closes #120 

---

## To review the changes

Use the "Files Changed" tab above.

## To review the full file tree

Do one of the following...

* [View it on GitHub](https://github.com/ourPLCC/plcc/tree/reorg).
* View it locally by cloning this project and switch to the `reorg` feature branch:
    ```bash
    git clone https://github.com/ourPLCC/plcc
    cd plcc
    git switch reorg
    ```

## Test it locally

1. Clone it locally and switch to the `reorg` branch (as shown above).
2. Add the full path to plcc/bin to your PATH and undefine LIBPLCC. Assuming you are in the root of the project:
    ```bash
    PATH="$(pwd)/src/plcc/bin:$PATH"
    LIBPLCC=
    ```
4. Try it out.
5. When you are done, exit the current terminal and things should go back to normal.

## Test it in GitPod

1. Open any project in GitPod.
2. Install PLCC from this PR's branch:
    ```bash
    PLCC_GIT_BRANCH=reorg /bin/bash -c "$(\curl -fsSL https://github.com/ourPLCC/plcc/raw/reorg/installers/plcc/install.bash)" >> ~/.bashrc
    ```
3. Open a new terminal.
4. Try it out.
5. When you are done, stop the GitPod workspace and throw it away.
